### ISSUE TITLE
"Tip-Adapter/main.py" use test features to eval

### ISSUE THREAD

**Issue by fikry102** [gaopengcuhk/Tip-Adapter#11]
It seems odd to use test features to eval.
see https://github.com/gaopengcuhk/Tip-Adapter/blob/fcb06059457a3b74e44ddb0d5c96d2ea7e4c5957/main.py#L111
Could authors give some explanation?

**Comment by waleedgondal** [comment]
I agree with @fikry102, best 'adapter' selection is done using test features. Shouldn't validation features be used for model selection?

**Comment by ZrrSkywalker** [comment]
@waleedgondal Thanks for pointing out. Sorry for this mistake. The model selection after every-epoch training should be rectified by using validation set. By our experiments, this would not affect the final performance on the test set.
